doc: Added overview documentation (#34)
eik-lib/eik-lib.github.io: docs/overview.md

title: Overview
sidebar_label: Overview
---

Eik is an [asset server](/docs/server) for serving ESM and CSS assets. The Eik server can run and be integrated with your own infrastructure. Eik comes with a [client](/docs/client) for easy upload and management of your assets.

> 

Reviewer: Do we want to mention that other assets (eg. SVG) can also be served or keep the scope narrow?

Author: Yeah. But I think that is more suitable in a more "extensive overview". We also need to explain the difference between the `npm` and `pkg` namespace.

Then we need to create a mapping between the bare import statement developers will use when developing the applications and the aliased URL of lit-html. We do so by making an Import Map as follow:

```json
{
    "lit-html": "/npm/lit-html/v1"
}
```

When created we publish this Import Map to our Eik server and we create an alias for it. If we named the Import Map "site-mapping" and versioned it as 1.0.1 during upload to the Eik server, it will now be available on the alias `https://eik-server.com/map/site-mapping/v1`.

In each application we can now depend on and install lit-html through NPM as many projects already do. Each application can now locally be developed by refering lit-html through its bare import statement:

```js
import * as lit from 'lit-html';
```

When an application is ready to be put into production, the frontend code should be uploaded as a package to the Eik server too. Upon preparation of our application to upload to the Eik server we will apply our Import Map. This will transform our bare import statements to a legal ESM import statement pointing to the lit-html alias defined in the Import Map:

```js
import * as lit from '/npm/lit-html/v1';
```

> Now our applications use a ESM import statement pointing to the alias for lit-html which makes sure multiple applications on our site align to the same version of lit-html. While doing so, we've also been able to develop our application in isolation without depending or interfering with other applications utlilizing the same library.

Reviewer: ```suggestion
Our application defines an ESM import statement that points to the alias for lit-html which makes sure multiple applications on our site align to the same version of lit-html. By doing this, we're able to develop our application in isolation without depending or interfering with any other applications that utilise the same library.
```
